### ISSUE TITLE
SMS and TEL Linking causes crash SDK 27

### ISSUE THREAD

**Issue by nikoootine** [expo/expo#1711]
### Environment

Environment:
  OS: macOS High Sierra 10.13.4
  Node: 9.11.1
  Yarn: Not Found
  npm: 5.6.0
  Watchman: 4.7.0
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: 3.1 AI-173.4670197

Packages: (wanted => installed)
  expo: ^27.0.1 => 27.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-27.0.0.tar.gz => 0.55.2

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/nupixo-09d50979-311b-4a7a-8c8e-3a78cd46f342.tar.gz
### Steps to Reproduce

1. Open this snack demo: https://snack.expo.io/@nikoootine/linking-bug-tel-sms
2. Press the "Press Me" button
3. See how the app restarts/crashes

### Expected Behavior
App doesnt crash. App will initiate a call or sms window

### Actual Behavior
App crashes. Throws a java error on a local build
![31408307_1948474298498369_1128298908070445056_n](https://user-images.githubusercontent.com/15661253/39403891-5a4d8cc4-4b3b-11e8-9376-4616dbb08001.jpg)

### Reproducible Demo

Snack demo: https://snack.expo.io/@nikoootine/linking-bug-tel-sms

**Comment by Promaethius** [comment]
I get the same error for android builds.

**Comment by boris-burgos** [comment]
+1

**Comment by esamelson** [comment]
Hi, thanks very much for reporting this issue. I've fixed it internally and the fix will go out the next time we do a minor release for SDK 27. I do want to note that it only affects SDK 27, so even if you have updated to the latest Expo client, these links will continue to work on SDK <=26.

**Comment by giautm** [comment]
I just upgraded to SDKv28 and facing this issues. Is this released yet? btw, can you tag issues with a label, ex: `SDK27 BUG`?

**Comment by esamelson** [comment]
Hi @giautm - sorry about this 😧it's not released yet, will keep this thread up-to-date when it is. The best indicator that this is an SDK 27 bug is just that the ticket is open, as we usually close tickets that are for older SDK versions once the new one is out.

**Comment by rogerroelofs** [comment]
Do we have a sense of when this will be released?  It also affects geo: urls.

**Comment by elliotstokes** [comment]
I'm also seeing this on `mailto` links. I assume the fix mentioned will fix all url types? Errors on canOpenURL and openURL. Same error as above.

**Comment by esamelson** [comment]
@rogerroelofs @elliotstokes yes, those links should be fixes as well. Apologies for the issue. Please watch our [blog](https://blog.expo.io/) -- we hope to announce the minor release in the next week or so.

**Comment by boris-burgos** [comment]
In our tests the issue is fixed.

in package.json set:

```
"expo": "^27.0.1",
"react-native": "https://github.com/expo/react-native/archive/sdk-27.0.2.tar.gz",
```

**Comment by KeabetsweFx** [comment]
Does this update also apply to Expokit?

**Comment by esamelson** [comment]
Hi @KeabetsweFx - yes, this update applies to ExpoKit as well. You'll need to follow the normal upgrade instructions [here](https://docs.expo.io/versions/v27.0.0/expokit/expokit#android) as well as making the one-time changes to your AndroidManifest.xml as noted in [this gist](https://docs.expo.io/versions/v27.0.0/expokit/expokit#android).
